### PR TITLE
CHW: WSI surface parameters encapsulated

### PR DESCRIPTION
**Pull request by vTurbine**
This removes implementation specific access to rendering surface dimensions.

### CODE REVIEW

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]
I'm not sure, that `SIZE` is Linux compatible. But even if it is, consider to use std::pair and C++17 syntax:
```cpp
class HW
{
	...
	std::pair<u32, u32> GetSurfaceSize() const;
	...
};
...

auto [width, height] = HW.GetSurfaceSize();
```

**Comment by Xottab-DUTY** [comment]
Yes, according to Travis CI, type `SIZE` is Windows-only, which is not suitable for us.

**Comment by vTurbine** [comment]
Expected smth like this. But saw GL renderer using ‘HDC’

**Comment by vTurbine** [comment]
The master is broken

File: src/Layers/xrRenderDX10/dx10HW.cpp [hunk 1]
```diff
@@ -446,3 +446,11 @@ void CHW::UpdateViews()
 
     _RELEASE(pDepthStencil);
 }
+
+std::pair<u32, u32> CHW::GetSurfaceSize() const
+{
+    return {
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderDX10/dx10HW.cpp]: `{` should be on the next line.

File: src/Layers/xrRenderGL/glHW.cpp [hunk 2]
```diff
@@ -187,3 +187,11 @@ HRESULT CHW::Present(UINT /*SyncInterval*/, UINT /*Flags*/)
     SDL_GL_SwapWindow(m_hWnd);
     return S_OK;
 }
+
+std::pair<u32, u32> CHW::GetSurfaceSize() const
+{
+    return {
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glHW.cpp]: `{` should be on the next line.

File: src/Layers/xrRenderDX9/dx9HW.cpp [hunk 3]
```diff
@@ -378,3 +378,16 @@ BOOL CHW::support(D3DFORMAT fmt, DWORD type, DWORD usage)
         return FALSE;
     return TRUE;
 }
+
+std::pair<u32, u32> CHW::GetSurfaceSize() const
+{
+    return {
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderDX9/dx9HW.cpp]: `{` should be on the next line.

**Review by Xottab-DUTY — APPROVED** [review]

**Pull request merged**